factory/bootstrap/bootstrapComponentsHandler.go

```diff
@@ -117,12 +119,25 @@ func (mbf *managedBootstrapComponents) RoundActivationHandler() process.RoundAct
 	return mbf.bootstrapComponents.roundActivationHandler
 }
 
+// SetShardCoordinator sets the shard coordinator
+func (mbf *managedBootstrapComponents) SetShardCoordinator(shardCoordinator sharding.Coordinator) error {
```

Comment:
do we need this set? I thought the bootstrap components creates the shard coordinator instance

Reply:
Only for tests, will change to remove this from the production managed bootstrap components